### ISSUE TITLE
Field value in Double format generates Int instead

### ISSUE THREAD

**Issue by ArcherEmiya05** [wuseal/JsonToKotlinClass#334]
I have this issue where a field holds a value of Double data type yet the generated was Int, probably it was due to the fact that some items and its field gives a whole number which is an Int data type. Meaning the field value is inconsistent with data types or at least format, no wonder since I already saw some of their bad structured and inconsistency. This is a third party provider so we cannot just request to fix those inconsistency on their side and will handle it instead on our side. So my question is how the generator decides what data type to use in this case, is it the number of times (frequency) it get int value vs double value?

**Comment by xurui1995** [comment]
Thanks for your feedback! 
I  just verified this issue.
Input json:
```
[
  {
    "key": 1.2
  },
  {
    "key": 1
  }
]
```
Generated Kotlin class:
```
class Test : ArrayList<TestItem>(){
    data class TestItem(
        val key: Int
    )
}
````

The potential reason could be the JsonObject of the Item generated by our code,  the value 1.2 was rewritten by 1 due to the same key, we will check that, thanks again!
Class : `ListClassGeneratorByJSONArray`
Mathod: `getFatJsonObject`

**Comment by ArcherEmiya05** [comment]
Same thing also happens with field value string/null instead of nullable String it uses Any is this intended?

**Comment by wuseal** [comment]
Fixed in 3.7.0
